pelican/plugins/obsidian/obsidian.py

```diff
@@ -98,10 +108,19 @@ def read(self, source_path):
         Parse content and metadata of markdown files.
         Also changes the links to the acceptable format for pelican
         """
+        with pelican_open(source_path) as text:
+            m = HEADER_RE.fullmatch(text)
+
+        if not m:
+            return super().read(source_path)
+
+        text = m.group("content")
```

Comment:
@arulrajnet Any chance you could explain why we need to do more or less the same thing here as we do in `YamlMetadataReader` and we can't use the contents of super directly? In other words, what is the edge case we solve for here? 

If there is a particular edge case here, can we add it to tests?

Reply:
Check this comment https://github.com/jonathan-s/pelican-obsidian/issues/12#issuecomment-2563904562

## The issue

The wiki links and wiki images are not working after adding `YamlMetadataReader` which is the main use case of this plugin.

For example

Markdown file, have the following content

```
![[pelican-in-rock.png]]
```
The expected output is, the image has to be found and loaded as HTML. 

But the actual output is `![pelican-in-rock.png]({static}/assets/images/pelican-in-rock.png)`. It's loaded as `md` not the HTML. 

## Root cause

The `super().read(source_path)` is converted the markdown to HTML. Then we replaced the wiki/obsidian links to markdown links in `self.replace_obsidian_links(content)`. 

That's why, we read the content as markdown `m.group("content")` as like `YamlMetadataReader` . Then replace the links. Then convert to HTML. 

The core future of this plugin is broken, and we fixed in this PR. 

## Other fixes

* This plugin is not works in Windows machine. Fixed that
* Added support for modern image formats `webp, avif` 
* Enabled logging.

Reply:
Added unit test for the content generation.  The following cases added

- Internal link
- Internal image
- Internal link but it doesn't exists

![image](https://github.com/user-attachments/assets/d043cc66-c678-4c01-bbea-b5e2d6df87fc)